astropy/__init__.py

```diff
@@ -318,17 +318,17 @@ def online_help(query):
     webbrowser.open(url)
 
 
-__dir__ = ['__version__', '__githash__', '__minimum_numpy_version__',
-           '__bibtex__', 'test', 'log', 'find_api_page', 'online_help',
-           'online_docs_root', 'conf']
+__dir_inc__ = ['__version__', '__githash__', '__minimum_numpy_version__',
+               '__bibtex__', 'test', 'log', 'find_api_page', 'online_help',
+               'online_docs_root', 'conf']
 
 
 from types import ModuleType as __module_type__
-# Clean up top-level namespace--delete everything that isn't in __dir__
+# Clean up top-level namespace--delete everything that isn't in __dir_inc__
```

Comment:
Must admit I find this auto-cleanup bad form, but that's somewhat out of scope of this PR!

Reply:
Not really a cleanup, I had to change the use of `__dir__` to list the attributes that shall not be deleted because setting a list to `__dir__` no more works.

Reply:
@saimn - I meant that I found the whole stanza rather bad form, not your change to it! It just seems that by being a bit more careful in the code itself, one can avoid introducing these unwanted things in the namespace (or delete them after their use). Anyway, as said, definitely out of scope here!